From 42f6161b64378643e984c51fa2ba26c9bf70ca67 Mon Sep 17 00:00:00 2001
From: Gurumoorthy Santhakumar <gsanthak@qti.qualcomm.com>
Date: Thu, 5 Feb 2026 10:33:35 +0530
Subject: [PATCH] driver: clk: NSSCC_QCE2204 config to enable nss clock

-Add CONFIG_CLK_QCOM_NSSCC_QCE2204 config
-Add disable clock ops in qcom clock framework

Change-Id: I896de7afa927478bfe51de76f8f245b2d78142b2
Signed-off-by: Gurumoorthy Santhakumar <gsanthak@qti.qualcomm.com>
---
 drivers/clk/qcom/Kconfig      | 8 ++++++++
 drivers/clk/qcom/Makefile     | 1 +
 drivers/clk/qcom/clock-qcom.h | 1 +
 3 files changed, 10 insertions(+)

diff --git a/drivers/clk/qcom/Kconfig b/drivers/clk/qcom/Kconfig
index f7eafbb1824..69eef58a8d2 100644
--- a/drivers/clk/qcom/Kconfig
+++ b/drivers/clk/qcom/Kconfig
@@ -47,6 +47,14 @@ config CLK_QCOM_IPQ5424
 	  on the Snapdragon IPQ5424 SoC. This driver supports the clocks
 	  and resets exposed by the GCC hardware block.
 
+config CLK_QCOM_NSSCC_QCE2204
+	bool "Qualcomm NSSCC QCE2204"
+	select CLK_QCOM
+	help
+	  Say Y here to enable support for the NSS Clock Controller
+	  on the QCE2204. This driver supports the clocks
+	  and resets exposed by the NSSCC hardware block.
+
 config CLK_QCOM_IPQ5210
 	bool "Qualcomm IPQ5210 GCC"
 	select CLK_QCOM
diff --git a/drivers/clk/qcom/Makefile b/drivers/clk/qcom/Makefile
index e56dc4ed3e1..59b04a67b47 100644
--- a/drivers/clk/qcom/Makefile
+++ b/drivers/clk/qcom/Makefile
@@ -9,6 +9,7 @@ obj-$(CONFIG_CLK_QCOM_APQ8096) += clock-apq8096.o
 obj-$(CONFIG_CLK_QCOM_IPQ4019) += clock-ipq4019.o
 obj-$(CONFIG_CLK_QCOM_IPQ5332) += clock-ipq5332.o
 obj-$(CONFIG_CLK_QCOM_IPQ5424) += clock-ipq5424.o
+obj-$(CONFIG_CLK_QCOM_NSSCC_QCE2204) += nsscc-qce2204.o
 obj-$(CONFIG_CLK_QCOM_IPQ5210) += clock-ipq5210.o
 obj-$(CONFIG_CLK_QCOM_IPQ9574) += clock-ipq9574.o
 obj-$(CONFIG_CLK_QCOM_IPQ9650) += clock-ipq9650.o
diff --git a/drivers/clk/qcom/clock-qcom.h b/drivers/clk/qcom/clock-qcom.h
index 5883adf328b..dbd808cbdd7 100644
--- a/drivers/clk/qcom/clock-qcom.h
+++ b/drivers/clk/qcom/clock-qcom.h
@@ -87,6 +87,7 @@ struct msm_clk_data {
 	const char * const		*dbg_rcg_names;
 
 	int (*enable)(struct clk *clk);
+	int (*disable)(struct clk *clk);
 	unsigned long (*set_rate)(struct clk *clk, unsigned long rate);
 };
 
-- 
2.34.1

